app/src/main/java/com/project/pv239/customtimealarm/api/DirectionsGetter.java

```diff
@@ -0,0 +1,50 @@
+package com.project.pv239.customtimealarm.api;
+
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+
+import com.project.pv239.customtimealarm.api.model.GoogleMapsApiResponse;
+import com.project.pv239.customtimealarm.database.entity.Alarm;
+import com.project.pv239.customtimealarm.helpers.converters.TrafficModelToString;
+import com.project.pv239.customtimealarm.helpers.converters.TravelModeToString;
+import com.project.pv239.customtimealarm.helpers.places.PlacesProvider;
+
+import retrofit2.Call;
+import retrofit2.Response;
+
+public class DirectionsGetter {
+    private GoogleMapsApi googleMapsApi;
+
+    public DirectionsGetter(){
+        this.googleMapsApi = new GoogleMapsApi();
+    }
+
+    public int getTimeToDestinationInSeconds(@NonNull Alarm alarm){
+        final Call<GoogleMapsApiResponse> responseCall = googleMapsApi.getService().getDirections(
+                PlacesProvider.getOrigin(),
+                alarm.getDestination(),
+                TravelModeToString.get(alarm.getTravelMode()),
+                alarm.getTimeOfArrival(),
+                TrafficModelToString.get(alarm.getTrafficModel()),
+                GoogleMapsApiKeyGetter.getApiKey()
+        );
+
+        try{
+            return new AsyncTask<Void, Void, Integer>(){
+                @Override
```

Comment:
async task ma byt static je to best practise vysvetlenie tu (vravel mi to aj on v linku je prikald ako to prerobit)
https://stackoverflow.com/questions/44309241/warning-this-asynctask-class-should-be-static-or-leaks-might-occur